db/changes.go

```diff
@@ -758,7 +776,10 @@ func (col *DatabaseCollectionWithUser) SimpleMultiChangesFeed(ctx context.Contex
 				if err != nil {
 					base.WarnfCtx(ctx, "Unable to obtain channel cache for %s, terminating feed", base.UD(chanName))
 					change := makeErrorEntry("Channel cache unavailable, terminating feed")
-					output <- &change
+					select {
+					case output <- &change:
+					case <-options.ChangesCtx.Done():
+					}
```

Comment:
We're already terminating the changes feed here, so I don't think we want to check the ChangesCtx, and we definitely don't want to potentially miss sending the error entry to output if it doesn't get picked by the select.

Reply:
I just saw your PR comment "Checking for the Done() channel is necessary in the case that the output receiver no longer exists".  Is that an actual possibility?  I don't think adding non-deterministic behaviour is needed if it's being defensive for a scenario that can't happen.

Reply:
It definitely can happen because the consumer in `GetChanges` will exit when the `optinos.ChangesCtx`. I've added code to remove `cancelCtx` from `GenerateChanges` since these were always the same.

As discussed offline, I would like to make the context only a parameter that is passed into the changes feed, but I think we should defer from this PR.